src/drivers/infiniband/hermon.c

```diff
@@ -3926,6 +4021,8 @@ static int hermon_probe ( struct pci_device *pci ) {
 	hermon->uar = pci_ioremap ( pci, uar,
 				    HERMON_UAR_NON_EQ_PAGE * HERMON_PAGE_SIZE );
 
+	hermon->physical_function = pci->busdevfn & 0x7;
```

Comment:
This is not a valid way to obtain the physical function number: see e.g. commit https://github.com/ipxe/ipxe/commit/76a7bfe93

Reply:
Okay, I can see that. I'm sure we can just drop this patch. It was included because I felt confident in lifting the change from Flexboot, and it seemed like someone might get some benefit from it. But it's not actually clear to me what cards this target.

Do you know if ConnectX-4 cards have multiple physfuncs?

Reply:
I dropped the `physical_function` patch in the newest re-roll.